signer/src/transaction_signer.rs

```diff
@@ -256,6 +285,87 @@ where
         Ok(true)
     }
 
+    #[tracing::instrument(skip(self))]
+    async fn handle_wsts_message(
+        &mut self,
+        msg: &message::WstsMessage,
+        bitcoin_chain_tip: &model::BitcoinBlockHash,
+    ) -> Result<(), error::Error> {
+        tracing::info!("handling message");
+        match &msg.inner {
+            wsts::net::Message::DkgBegin(_) => {
+                let state_machine = self.create_new_state_machine()?;
+                self.wsts_state_machines.insert(msg.txid, state_machine);
+                self.relay_message(msg.txid, &msg.inner, bitcoin_chain_tip)
+                    .await?;
+            }
```

Comment:
Do we have to worry about what happens during a flash block here? I couldn't convince myself one way or the other.

Reply:
Unless I've done a mistake somewhere, the whole message passing procedure for a DKG round or a signing round should reuse the same chain tip. I.e. if a signer receives a nonce request for a particular chain tip, the nonce response will have the same chain tip even if the signer state is updated in the meantime.

This way, flash blocks shouldn't be able to cause any corruption or inconsistencies for any signing rounds or DKG rounds.

However, depending on the details of the validation we'll implement in #297 flash blocks may or may not cause signing rounds or DKG rounds to halt. Currently I don't think we should reject requests with stale chain tips, but we may want to include some parameter for rejecting messages with very old chain tips. A single flash block should probably never trigger that logic, but a block storm on testnet might.

In either case, we should be careful when we implement the rest of the system such that the worst thing that can happen is a failed signing round - which the system should tolerate.

Reply:
Ahh, I see. Thank you for explaining.